pkg/loader/tf.go

```diff
@@ -0,0 +1,700 @@
+// Copyright 2021 Fugue, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loader
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/terraform/configs"
+	"github.com/zclconf/go-cty/cty"
+
+	"tf_resource_schemas"
+)
+
+type TfDetector struct{}
+
+func (t *TfDetector) DetectFile(i InputFile, opts DetectOptions) (IACConfiguration, error) {
+	if !strings.HasSuffix(i.Path(), ".tf") {
+		return nil, fmt.Errorf("Expected a .tf extension for %s", i.Path())
+	}
+
+	return parseFiles([]string{}, nil, []string{i.Path()})
+}
+
+func (t *TfDetector) DetectDirectory(i InputDirectory, opts DetectOptions) (IACConfiguration, error) {
+	// First check that a `.tf` file exists in the directory.
+	if matches, err := filepath.Glob(i.Path() + "/*.tf"); err != nil || len(matches) == 0 {
```

Comment:
Another way to do this would be to loop through `i.Children()` and if the child's type is `InputFile` it has an `Ext()` method that will return `.tf` for TF files. It would cut down on disk IO since we've already extracted that info at this point.